test/integration/controller/jobs/raycluster/raycluster_webhook_test.go

```diff
@@ -56,10 +66,95 @@ var _ = ginkgo.Describe("RayCluster Webhook", func() {
 		})
 
 		ginkgo.It("the creation doesn't succeed if the queue name is invalid", func() {
-			job := testingjob.MakeCluster(jobName, ns.Name).Queue("indexed_job").Obj()
+			job := testingraycluster.MakeCluster(jobName, ns.Name).Queue("indexed_job").Obj()
 			err := k8sClient.Create(ctx, job)
 			gomega.Expect(err).Should(gomega.HaveOccurred())
 			gomega.Expect(apierrors.IsForbidden(err)).Should(gomega.BeTrue(), "error: %v", err)
 		})
 	})
+
+	ginkgo.When("With manageJobsWithoutQueueName enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				CRDPath:     crdPath,
+				DepCRDPaths: []string{rayCrdPath},
+				WebhookPath: webhookPath,
+			}
+			cfg = fwk.Init()
+			ctx, k8sClient = fwk.RunManager(cfg, managerWithRayClusterAndRayJobControllersSetup(jobframework.WithManageJobsWithoutQueueName(true)))
+		})
+		ginkgo.BeforeEach(func() {
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "raycluster-",
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		})
+		ginkgo.AfterAll(func() {
+			fwk.Teardown()
+		})
+
+		ginkgo.It("Should not suspend a cluster if the parent's workload exist and is admitted", func() {
+			ginkgo.By("Creating the parent job which has a queue name")
+			parentJob := testingrayjob.MakeJob("parent-job", ns.Name).
+				Queue("test").
+				Suspend(false).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, parentJob)).Should(gomega.Succeed())
+
+			lookupKey := types.NamespacedName{Name: parentJob.Name, Namespace: ns.Name}
+			gomega.EventuallyWithOffset(1, func() error {
+				if err := k8sClient.Get(ctx, lookupKey, parentJob); err != nil {
+					return err
+				}
+				parentJob.Status.JobDeploymentStatus = rayv1alpha1.JobDeploymentStatusSuspended
+				return k8sClient.Status().Update(ctx, parentJob)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
```suggestion
```suggestion
			gomega.Eventually(func(g gomega.Gomega) {
				g.Expect(k8sClient.Get(ctx, lookupKey, parentJob)).Should(gomega.Succeeded())
				parentJob.Status.JobDeploymentStatus = rayv1alpha1.JobDeploymentStatusSuspended
				g.Expect(k8sClient.Status().Update(ctx, parentJob)).Should(gomega.Succeeded())
			}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Any offset isn't needed, right?

Reply:
Right, updated in e2cee02b33adec66e756c9a5640ca6b2276a9152.